### DISCUSSION TITLE
HASS service call to reload MASS player providers

### DISCUSSION THREAD

**Discussion by miumer** [music-assistant/support#2882] [Q&A] [answered]
My HASS player provider players seem to disconnect from MASS sometimes when i restart my computer, my containers etc. They start working when i reload the HASS player provider, but i have to go into MASS UI for this. Would like to make a service call via automation to reload if necessary.

**Comment by sp0ngeboobs** [comment]
I have the same problem when there is a power outage (Home Assistant has backup power) or a router reboot, media players are not available in MA, you need to go to the menu and reboot the provider. I partially solved this issue with automation - if the media player is unavailable for 5 minutes, then I reload the MA addon. But it doesn't always work.

**Reply by OzGav** [reply]
Retry logic was added some time ago to fix something similar https://github.com/music-assistant/hass-music-assistant/issues/2256

How long are you waiting for the player to reappear? Is the HA plugin showing as working ok? Is the HA provider showing as working ok? What HA integration is supplying the player?

**Reply by miumer** [reply]
For me the player provider seems fine when looking at it in MA UI, but when i go to HA -> MASS integration -> devices, all the players Status is RED (!). When i reload the MASS integration in HASS it does not help. Only when i go into the MASS UI and reload the HASS player provider the players Status turn to OK in HASS. Im using chrome-cast and also VLC-Telnet in HASS as players. Both behave the same.

Would not be a problem if only reloading the MASS integration would work. HASS has that service allready.

**Reply by OzGav** [reply]
OK for the Chromecast players you should be using the native MA player provider. @marcelveldt what do you think about the VLC player not appearing as available?

**Comment by miumer** [comment]
Switched to squeezelite. Same thing is happening with squeezelite now. I need to reload it everytime i restart my host machine (or when it wakes up from sleep even). It might be that the squeezelite service loads after the MASS docker image or smth. Point is id like to be able to run a HASS service call to "reload a player provider" in MASS so i dont have to do it manually. 

Especially since i cant access MASS from companion devices, which i wish was also possible (without reverse proxy), but i guess is harder to achiveve because of http. So basically i have to open my host device with a display to reload the provider..

**Reply by OzGav** [reply]
Something else is going on with your setup if you are saying you are using the slimproto provider in MA and the player is going unavailable after a restart. No on else is reporting that. It could be a device problem. I use picoreplayers and have no issues.

**Comment by TechMoeve** [comment]
I have some mobil boxes. If i switch them on, i have to wait a long time until they are available in mass. For this usecase it would be nice to have the possibility to reload the player provider by an automation.

**Reply by OzGav** [reply]
What is a "mobil box"?

**Reply by TechMoeve** [reply]
I meant a mobile box, in this case for example a sonos roam in my bathroom. Most times its switched off and as a result not available in music assistant. If i go to the bathroom and switch it on, it needs a long time, until the box is available. As a workaround i have an automation which restarts the whole addon:


```
triggers:
 - trigger: state
   entity_id:
      - media_player.sonos_mobil
      from: unavailable
      for:
        hours: 0
        minutes: 0
        seconds: 5
conditions: []
actions:
 - action: hassio.addon_restart
   metadata: {}
   data:
      addon: xxxxxxx_music_assistant
```

After the restart the box is available in music assistant. But it would be nice just to reload the sonos provider in an automation, cause restarting the addon stops the other mediaplayers.

**Reply by OzGav** [reply]
This was previously reported here and was reported as fixed by the OP https://github.com/music-assistant/support/issues/2819

**Reply by marcelveldt** [reply]
If mdns is working properly, a device back online should be discovered within seconds. I use that myself as well, several sonos speakers just get unplugged from power and when I plug one back in power, it gets rediscovered within 30 seconds.

**Reply by TechMoeve** [reply, marked as answer]
Indeed it works. I think i didn´t try it the last weeks cause of my workaround. Very nice!
